packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/notifications/BacklinksNotificationsTests.kt

```diff
@@ -408,4 +409,242 @@ class BacklinksNotificationsTests : RealmEntityNotificationTests {
+        }
+        assertIs<InitialResults<Sample>>(c.receiveOrFail())
+        realm.write {
+            // Update field that should not trigger a notification
+            findLatest(results.first())!!
+                .objectBacklinks.first()
+                .objectBacklinks.first()
+                .objectBacklinks.first()
+                .objectBacklinks.first()
+                .objectBacklinks.first()
+                .stringField = "Bar"
+        }
+        realm.write {
+            // Update field that should trigger a notification
+            findLatest(results.first())!!.intField = 1
+        }
+        c.receiveOrFail().let { resultsChange ->
+            assertIs<UpdatedResults<Sample>>(resultsChange)
+            when (resultsChange) {
+                is ResultsChange<*> -> {
+                    // Default value is 42, so if this event is triggered by the first write
+                    // this assert will fail
+                    assertEquals(1, resultsChange.list.first().intField)
+                }
+                else -> fail("Unexpected change: $resultsChange")
+            }
+        }
+        observer.cancel()
+        c.close()
+    }
+
+    @Test
+    override fun keyPath_propertyBelowDefaultLimit() = runBlocking<Unit> {
+        val c = Channel<ResultsChange<Sample>>(1)
+        realm.write {
+            copyToRealm(
+                Sample().apply {
+                    this.intField = 1
+                    this.stringField = "parent"
+                    this.nullableObject = Sample().apply {
+                        this.stringField = "child"
+                        this.nullableObject = Sample().apply {
+                            this.stringField = "child-child"
+                            this.nullableObject = Sample().apply {
+                                this.stringField = "child-child-child"
+                                this.nullableObject = Sample().apply {
+                                    this.stringField = "child-child-child"
+                                    this.nullableObject = Sample().apply {
+                                        this.stringField = "BottomChild"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            )
+        }
+        val results = realm.query<Sample>("stringField = 'BottomChild'").find()
+        val observer = async {
+            results.asFlow(listOf("objectBacklinks.objectBacklinks.objectBacklinks.objectBacklinks.objectBacklinks.stringField")).collect {
+                c.trySend(it)
+            }
+        }
+        assertIs<InitialResults<Sample>>(c.receiveOrFail())
+        realm.write {
+            // Update field that should not trigger a notification
+            findLatest(results.first())!!.booleanField = true
+        }
+        realm.write {
+            // Update field that should trigger a notification
```

Comment:
Yes, because I used the keypath `listOf("objectBacklinks.objectBacklinks.objectBacklinks.objectBacklinks.objectBacklinks.stringField")` which only consider changes through the backlinks as things that should trigger a notification